### PR TITLE
more config specs cleanup & remove deprecated from knife config list

### PR DESCRIPTION
**Pull request by vsingh-msys**
## Description

We have let with some cleanup in https://github.com/chef/chef/pull/10347 for PR https://github.com/chef/chef/pull/10346

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).
Signed-off-by: Vivek Singh <vivek.singh@msystechnologies.com>

### CODE REVIEW

File: spec/integration/knife/config_use_spec.rb [hunk 1]
```diff
@@ -67,6 +67,54 @@
...
+        end
+
+        it { is_expected.to eq "other\n" }
+      end
+
+      context "with a context file under $KNIFE_HOME" do
+        before do
+          file("knifehome/.chef/context", "other\n")
+          ENV["KNIFE_HOME"] = path_to("knifehome")
+        end
```
Comment (phiggins) [inline spec/integration/knife/config_use_spec.rb:112]: Do these environment variables need to be reset after the test runs?

**Comment by tas50** [comment]
You might want to look at how we handled deprecated knife plugins without duplicating the code in the past:

https://github.com/chef/chef/pull/9288/files#diff-4a6956aeb8c58095046db0d012e962f0L1

File: spec/integration/knife/config_use_spec.rb (hunk 1 above)
Comment (vsingh-msys) [reply spec/integration/knife/config_use_spec.rb:112]: Probably Yes, seems we are storing the old value of these environments variables and resetting back it to the original one.

Copy-paste stuff I haven't check in detail :smile:

**Comment by vsingh-msys** [comment]
@tas50 thanks, that's looked cleaner way deprecated the options

File: lib/chef/knife/config_list.rb [hunk 2]
```diff
@@ -21,9 +21,8 @@ class Chef
   class Knife
     class ConfigList < Knife
       banner "knife config list (options)"
-      category "deprecated"
 
-      TABLE_HEADER = [" Profile", "Client", "Key", "Server"].freeze
+      TABLE_HEADER = [" Profile", "Client", "Key", "Server"].freeze unless const_defined?(:TABLE_HEADER)
```
Comment (vsingh-msys) [inline lib/chef/knife/config_list.rb:25]: Getting already initialized warning suppress it by checking it's already defined or not.

Comment (tas50) [reply lib/chef/knife/config_list.rb:25]: We really need to track down why this happens, but the easiest way to get around it is ||= instead of =

**Comment by tas50** [comment]
@vsingh-msys Can you take a look at the spec failures due to formatting here:

https://buildkite.com/chef/chef-chef-master-omnibus-release/builds/758#ea0bd459-a0fa-4d28-b25e-85320054f1ab/5588-6838

File: lib/chef/knife/config_list.rb (hunk 2 above)
Comment (vsingh-msys) [reply lib/chef/knife/config_list.rb]: seems it's an issue with all knife commands, used `||=` for now. Thanks!

File: lib/chef/knife/config_list.rb [hunk 3]
```diff
@@ -128,8 +127,7 @@ def render_table(profiles, padding: 1)
         else
           table.render do |renderer|
             renderer.border do
-              mid   "-"
-              style :green
```
Comment (vsingh-msys) [inline lib/chef/knife/config_list.rb:132]: Somehow `TTY::Table` unable to add border-color for some OS. Keeping plain border here.

**Comment by tas50** [comment]
Thanks for the cleanup @vsingh-msys

**Pull request merged**